### PR TITLE
updating df.query and df.eval docstrings. resolves #16283

### LINKED ISSUE #16283: df.eval does not see globals
#### Code Sample, a copy-pastable example if possible

```python
from numpy import pi
import pandas as pd

df = pd.DataFrame({'a': [1, 2], 'b': [3, 4]})

print('pi' in globals())
print(df.eval('a * b * pi'))
```
#### Problem description
This raises a 
```
pandas.computation.ops.UndefinedVariableError: name 'pi' is not defined
```

According to the documentation of `pd.eval` which is used by `pd.DataFrame.eval`, 
globals should be available:

> local_dict : dict or None, optional
A dictionary of local variables, taken from locals() by default.
global_dict : dict or None, optional
A dictionary of global variables, taken from globals() by default.

The same exception is raised even if `global_dict=globals()` is passed to eval explicitly.

The only solution I found was to use:

```python
pd.eval('a * b * pi', resolvers=[df, globals()])
```

#### Expected Output
No exception and the result of the calculation

#### Output of ``pd.show_versions()``

<details>
# Paste the output here pd.show_versions() here

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.8.0-51-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 35.0.1
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
statsmodels: 0.8.0
xarray: None
IPython: 6.0.0
sphinx: 1.4.8
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.0
tables: 3.3.0
numexpr: 2.6.2
matplotlib: 2.0.1
openpyxl: 2.4.1
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.5.3
html5lib: 0.999
httplib2: 0.9.2
apiclient: None
sqlalchemy: 1.1.9
pymysql: 0.7.2.None
psycopg2: None
jinja2: 2.9.6
boto: 2.43.0
pandas_datareader: Non
</details>

### PR DESCRIPTION
**Pull request by mutricyl**
- [x] closes #16283
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This is a docstring improvement suggested in #16283 issue discussion. `df.eval` and `df.query` are updated for consistency and adding exemples with `@` usage for local variables

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4644,6 +4652,20 @@ def eval(self, expr: str, *, inplace: bool = False, **kwargs) -> Any | None:
...
+
+            You can refer to column names that are not valid Python variable names
+            by surrounding them in backticks. Thus, column names containing spaces
+            or punctuations (besides underscores) or starting with digits must be
+            surrounded by backticks. (For example, a column named "Area (cm^2)" would
+            be referenced as ```Area (cm^2)```). Column names which are Python keywords
+            (like "list", "for", "import", etc) cannot be used.
+
+            For example, if one of your columns is called ``a a`` and you want
+            to sum it with ``b``, your query should be ```a a` + b``.
```
Comment (mroeschke) [inline pandas/core/frame.py:4668]: Can you demonstrate this in the `Example` section instead?

Comment (mutricyl) [reply pandas/core/frame.py:4668]: I have added one example in the Example section. Note that above wording is coming from `query` function.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -4695,19 +4719,19 @@ def eval(self, expr: str, *, inplace: bool = False, **kwargs) -> Any | None:
         modified.
 
         >>> df.eval("C = A + B")
-           A   B   C
-        0  1  10  11
-        1  2   8  10
-        2  3   6   9
-        3  4   4   8
-        4  5   2   7
+           A   B  C C   C
```
Comment (mroeschke) [inline pandas/core/frame.py:4722]: These headers don't look correct anymore

File: pandas/core/frame.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/frame.py:4668]: I would remove all this explanation now that it's explained in the examples section

File: pandas/core/frame.py (hunk 2 above)
Comment (mutricyl) [reply pandas/core/frame.py:4715]: I double checked and it looks correct

Comment (mroeschke) [reply pandas/core/frame.py:4715]: If you look at the examples here you can see 5 column labels for 4 columns: https://pandas.pydata.org/preview/pandas-dev/pandas/58749/docs/reference/api/pandas.DataFrame.eval.html

Comment (mutricyl) [reply pandas/core/frame.py:4715]: Initial example dataframe contains 3 columns: `A`, `B` and `C C`. The eval method will add a fourth column named `C` in the resulting dataframe for this example.
Repr of the dataframe does not make the separation of columns obvious when columns contains spaces. It is like there is 3 columns `C` when you actually have `C C` and `C`.

To make the example cristal clear I may change `C C` by `C&C` and update the examples to add columns with names differents than `C`.

Do you want me to update also `query` examples as `C C` is actually coming from there ??

Comment (mroeschke) [reply pandas/core/frame.py:4715]: > To make the example cristal clear I may change C C by C&C and update the examples to add columns with names differents than C.

Ah yes that would be great. Sorry I didn't backtrack to see the context of the original columns

Comment (mutricyl) [reply pandas/core/frame.py:4715]: shall I update `query` as well ?

Comment (mroeschke) [reply pandas/core/frame.py:4715]: Yes please

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @mutricyl

**Pull request merged**